src/lib/update-alternatives/UI/main_dialog.rb

```diff
@@ -74,6 +91,27 @@ def map_alternatives_items
       end
     end
 
+    def filtered_alternatives
+      alternatives = @alternatives_list.select { |a| Regexp.new(@search).match(a.name) }
```

Comment:
Well, I think that use regexp is correct, otherwise there should be write the exact name of an alternative.